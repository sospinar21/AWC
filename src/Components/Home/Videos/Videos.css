.videos-main {
  display: flex; }

.video-card {
  margin: 1vh 15%; }

.selectedVideo {
  width: 80%;
  padding-top: 8vh;
  text-align: center; }

iframe {
  height: 460px;
  width: 840px; }

.videosidebar {
  width: 30%;
  height: 99vh;
  background-color: white;
  overflow: scroll;
  -webkit-box-shadow: 1px 4px 11px lightgrey;
  box-shadow: 1px 4px 11px lightgrey; }

.video-description {
  text-align: left;
  display: -ms-flexbox;
  display: flex;
  width: 80%;
  margin: 2vh auto;
  background-color: #a9a9a91c;
  padding: 4vh;
  box-sizing: border-box;
  box-shadow: 0px 0px 6px #d3d3d394; }
  .video-description h3 {
    font-size: 30px;
    color: black; }
  .video-description .icons {
    display: flex;
    width: 25%; }
  .video-description .icon-vid {
    font-size: 4vh;
    margin: 0 1vh; }
