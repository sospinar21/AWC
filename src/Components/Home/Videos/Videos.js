import React, { Component } from 'react';
import  ApiCalls  from '../../../Helper/ApiCalls/ApiCalls';
import { connect } from 'react-redux';
import { addVideos } from '../../../Actions/actions';
import './Videos.css';
import PropTypes from 'prop-types';

export class Videos extends Component {
  constructor() {
    super() 

    this.state = {
      selectedVideo: '',
      selectedTitle:''
    }
  }

  componentDidMount() {
    this.fetchVideos();
  }

fetchVideos = async () => {
  let api = new ApiCalls();
  const suggestedVideos = await api.fetchVideos();
  await this.props.addVideos(suggestedVideos);

}

displaySelectedVideo = (id, title) => {
  console.log(title)
 if (id){
  this.setState({
    selectedVideo: id,
    selectedTitle: title
  })
 }
}

displayVideos = () => {
  const videos = this.props.suggestedVideos.map(video => {
    const snippet = video.snippet.thumbnails.medium.url;
    const title = video.snippet.title;
    const videoId = video.id.videoId;
    return (
      <div onClick={() => this.displaySelectedVideo(videoId, title)} key={videoId} className='video-card'>
        <img src={snippet} />
      </div>
    );
  });

  return videos;
}


render () {
  const selectedVideo = this.state.selectedVideo.length ? this.state.selectedVideo : 'Kl5B6MBAntI'
  const selectedTitle = this.state.selectedTitle.length ? this.state.selectedTitle: 'ED SHEERAN - Shape Of You | Kyle Hanagami Choreography'

  if (this.props.suggestedVideos.length) {
    return (
      <div className='videos-main'>
        <div className='selectedVideo'> 
          <div key={selectedVideo} className='video-display'>
            <iframe src={`https://www.youtube.com/embed/${selectedVideo}`} />
          </div>
          <div className='video-description'>
            <h3>{selectedTitle}</h3>
            <div className='icons'>
              <i className="material-icons icon-vid">thumb_up_alt</i> 
              <p>0</p>
              <i className="material-icons icon-vid">thumb_down_alt</i> 
              <p>0</p>
            </div>
          </div>
        </div>
        <div className='videosidebar'>
          {this.displayVideos()}
        </div>
      </div>
    );
      
  } else {
    return (
      <h1>nadieata</h1>
    );
  }
}
}

export const mapStateToProps = (state) => {
  return ({
    suggestedVideos: state.suggestedVideos
  });
};

export const mapDispatchToProps = dispatch => ({
  addVideos: (VideosData) => dispatch(addVideos(VideosData)) 
  
});
export default connect(mapStateToProps, mapDispatchToProps)(Videos);