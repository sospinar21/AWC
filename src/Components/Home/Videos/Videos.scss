.videos-main {
    display: flex;
}

.video-card {
  margin: 1vh 15%;
}

.selectedVideo {
  width: 80%;
  padding-top: 8vh;
  text-align: center;
}

iframe {
  height: 460px;
  width: 840px;
}

.videosidebar {
  width: 30%;
  height: 99vh;
  background-color: white;
  overflow: scroll;
  -webkit-box-shadow: 1px 4px 11px lightgrey;
  box-shadow: 1px 4px 11px lightgrey;
}

.video-display {

}

.video-description {
  text-align: left;
  display: -ms-flexbox;
  display: flex;
  width: 80%;
  margin: 2vh auto;
  background-color: #a9a9a91c;
  padding: 4vh;
  box-sizing: border-box;
  box-shadow: 0px 0px 6px #d3d3d394;

  h3 {
    font-size: 30px;
    color: black
  }

  .icons {
    display: flex;
    width: 25%;
  }

  .icon-vid{
    font-size: 4vh;
    margin: 0 1vh;
  }

}