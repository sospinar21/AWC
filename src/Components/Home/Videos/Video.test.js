import {Videos, mapStateToProps, mapDispatchToProps} from '../Videos/Videos'
import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from 'enzyme';

describe('videos', () => {

  let videos
  let mockProps

  beforeEach(() => {

    mockProps = {
      suggestedVideos: [{id: 'dance'}],
      addVideos: jest.fn()
    }

    videos = shallow(<Videos  {...mockProps}/> , { disableLifecycleMethods: true })

  })

  it('matches snapshot', () => {
    expect(videos).toMatchSnapshot();
  })
})