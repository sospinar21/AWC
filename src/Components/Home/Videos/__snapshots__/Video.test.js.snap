// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`videos matches snapshot 1`] = `undefined`;
